### DISCUSSION TITLE
E9007 Error When Updating nanoCLR on ESP32

### DISCUSSION THREAD

**Discussion by homostultus39** [nanoframework/Home#1400] [Q&A] [answered]
I encountered an issue while updating nanoCLR on my ESP32 device and received error E9007. I provide the results and details about my device below:

C:\Windows\System32>nanoff --update --clrfile "C:\nanofw\nanoCLR.bin" --serialport COM3 --target ESP32
.NET nanoFramework Firmware Flasher v2.5.46+527e3d0ff4
Copyright (C) 2019 .NET Foundation and nanoFramework project contributors

Reading details from chip...OK

Connected to:
ESP32 (ESP32-D0WD-V3 (revision v3.1))
Features WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Flash size 4MB unknown from TENX (manufacturer 0x94 device 0x16406)
PSRAM: undetermined
Crystal 40MHz
MAC 24:DC:C3:9B:B7:B8

E9007

Could you suggest how to solve this problem?

**Comment by josesimoes** [comment]
Error when flashing the device. 
https://github.com/nanoframework/nanoFirmwareFlasher/blob/527e3d0ff428a3a4cce39b6f147c3e205f5958f6/nanoFirmwareFlasher.Library/ExitCodes.cs#L292

Better use `--platform esp32` instead of `--target ESP32`

**Reply by homostultus39** [reply]
I replaced --target with --platform, the problem remained. Or should I have done something else?

**Reply by josesimoes** [reply]
The output is the same?

**Reply by homostultus39** [reply]
one more line added to the output:
No target name was provided! Using 'ESP32_REV3' based on the device characteristics.
E9007

**Reply by josesimoes** [reply]
Suggest that you check the cable. 
And/or maybe add `-v diag` to grab a verbose output on the operations.

**Reply by homostultus39** [reply]
I changed the cable, tried again, nothing changed. added v -diag, got the following output:
C:\Windows\System32>nanoff --update --clrfile "C:\nanofw\nanoCLR.bin" --serialport COM3 --platform esp32 -v diag
.NET nanoFramework Firmware Flasher v2.5.46+527e3d0ff4
Copyright (C) 2019 .NET Foundation and nanoFramework project contributors


Using COM3 @ 1500000 baud to connect to ESP32.
Reading details from chip...
Executing esptool with the following parameters:
'--port COM3  --chip auto   --after no_reset_stub flash_id'

OK

Connected to:
ESP32 (ESP32-D0WD-V3 (revision v3.1))
Features WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Flash size 4MB unknown from TENX (manufacturer 0x94 device 0x16406)
PSRAM: not available
Crystal 40MHz
MAC 24:DC:C3:9B:B7:B8


No target name was provided! Using 'ESP32_REV3' based on the device characteristics.

Error E9007: Error downloading firmware file.

**Reply by josesimoes** [reply]
You may have issues accessing the cloudsmith repository... are you behind any corporate firewall or your ISP is preventing you to access some IP addresses?

**Reply by homostultus39** [reply]
Most likely it's an ISP issue

**Reply by homostultus39** [reply, marked as answer]
Everything worked out, it turns out that you just needed to install a VPN. Thank you very much!
